src/Mod/Import/gltfGlbExt.py

```diff
@@ -0,0 +1,371 @@
+# -*- coding: utf8 -*-
+"""GLTF Export Extension Module for FreeCAD.
+
+This module provides fixes for a material naming issue with FreeCAD's default glTF exporter,
+as well as exports some useful metadata by parsing VarSets.
+"""
+
+import ImportGui
+import json
+import struct
+
+
+def collect_all_material_names(objects):
+    """Collect unique material names from object hierarchy in discovery order"""
+
+    visited = set()
+
+    def extract_materials(obj):
+        obj_id = id(obj)
+        if obj_id in visited:
+            return
+        visited.add(obj_id)
+
+        try:
+            for attr_name in dir(obj):
+                if attr_name.startswith("__"):
+                    continue
+
+                try:
+                    attr_value = getattr(obj, attr_name)
+                    if callable(attr_value):
+                        continue
+
+                    if (
+                        attr_name in ["Material", "ShapeMaterial"]
+                        and attr_value
+                        and hasattr(attr_value, "Name")
+                    ):
+                        material_name = str(attr_value.Name)
+                        if material_name not in found_material_names:
+                            found_material_names.append(material_name)
+
+                    elif hasattr(attr_value, "__dict__") and not isinstance(
+                        attr_value, (str, int, float, bool, list, tuple, dict)
+                    ):
+                        extract_materials(attr_value)
+
+                except (AttributeError, TypeError, ValueError):
+                    continue
+        except (AttributeError, TypeError):
```

Comment:
On this exception clause, and all others that follow this format, you should either include text explaining why it's safe to simply ignore these, or add logging output for diagnostic purposes, or both.